rechunker/api.py

```diff
@@ -309,6 +301,7 @@ def _rechunk_array(
                 f"Got array_dims {array_dims}, target_chunks {target_chunks}."
             )
 
+    # TODO: rewrite to avoid the hard dependency on dask
```

Comment:
I think we could easily vendor this function.

https://github.com/dask/dask/blob/cd2a42b1735dbd0eb66b95b2989047a7f02bbe50/dask/utils.py#L1185-L1236

Reply:
I agree!

I didn't do that yet because removing the hard dask dependency felt orthogonal to the pain point of this PR. But it's definitely worth doing soon...